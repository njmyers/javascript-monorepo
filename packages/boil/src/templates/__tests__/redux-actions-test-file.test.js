import reduxActionsTestFile from "../redux-actions-test-file";

describe("it works", () => {
  test("it is a function ", () => {
    expect(typeof reduxActionsTestFile).toBe("function");
  });

  test("it runs without crashing", () => {
    expect(reduxActionsTestFile("test-component")).toMatchSnapshot();
  });
});
