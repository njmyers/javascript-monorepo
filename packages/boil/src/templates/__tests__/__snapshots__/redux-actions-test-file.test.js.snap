// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works it runs without crashing 1`] = `
"import configureStore from 'redux-mock-store';
import * as testComponentActions from '../test-component-actions'
import { initialState } from '../test-component-reducer'

const middlewares = [];
const mockStore = configureStore(middlewares);
const store = mockStore(initialState)

describe('it creates actions', () => {
  beforeEach(() => {
    store.clearActions();
  });

  test('it is a function type', () => {
    Object.values(testComponentActions).forEach((action) => {
      expect(typeof action).toBe('function')
    });
  });

  test('it returns a plain object with a type property', () => {
    Object.values(testComponentActions).forEach((action) => {
      expect(action()).toHaveProperty('type')
    });
  });

  Object.values(testComponentActions).forEach((action) => {
    const type = action().type

    test(\`it dispatches \${type} type action from a mock store\`, () => {
      store.dispatch(action())
      const actions = store.getActions();

      expect(actions[0].type).toBe(type)
    });
  });
})
"
`;
